### DISCUSSION TITLE
Mainnet: precheck failed: maxPriorityFeePerGas is 1000000 but must be at least 1250000000

### DISCUSSION THREAD

**Discussion by kowalski** [alchemyplatform/aa-sdk#854] [Q&A] [answered]
Thanks for filling out this bug report. Please provide as much detail as possible.

If your issue is with our RPC APIs and not the SDK please report your issue on (discord)[https://alchemy.com/discord].

### [REQUIRED] Environment

- AA SDK version: 3.18.2
- Package: @alchemy/aa-core

#### Relevant code or sample repro:

I'm using a off the shelf client which I construct as:

```
import { createSmartAccountClient as createSmartAccountClientAlchemy,} from '@alchemy/aa-core'

...

  return createSmartAccountClientAlchemy({
    transport: config.transport,
    chain: config.chain,
    account: config.account,
    paymasterAndData: paymasterConfig && {
      paymasterAndData: async (struct) =>
        paymasterConfig.paymasterAndData
          ? {
              ...struct,
              paymasterAndData: paymasterConfig.paymasterAndData,
            }
          : struct,
      dummyPaymasterAndData: () => paymasterConfig.dummyPaymasterAndData,
    },
    opts: {
      txMaxRetries: 10,
    },


```

When trying to send a transaction on mainnet I'm always getting error like:
```
precheck failed: maxPriorityFeePerGas is 1000000 but must be at least 1250000000
```
I've put console.logs in `node_modules/@alchemy/aa-core/dist/cjs/middleware/defaults/feeEstimator.js`
file and what I can tell is that the line
```
    const feeData = await client.estimateFeesPerGas();
```
returns a feeData which has `maxPriorityFeePerGas` set to 1000000. 

I'm confused about where the requirements of putting at least 1250000000 comes from? Is it on the bundler API level? How can I learn what the minimum acceptable value is?

**Comment by moldy530** [comment, marked as answer]
if you're using alchemy's RPCs and a custom paymaster, as it seems you're doing here, you should also import the `alchemyFeeEstimator` from `@alchemy/aa-alchemy` to use here instead as that will get the right fee estimate

**Reply by kowalski** [reply]
Thanks for this suggestion. We will give it and try and let you know if it works

**Reply by kowalski** [reply]
@siepra FYI

**Comment by siepra** [comment]
@moldy530 `alchemyFeeEstimator` seems to do the trick, thank you! I'll let you know shall we encounter any other issues in the matter

**Reply by avarobinson** [reply]
Awesome! Let us know if you run in to other issues 🙏
